lib/plugins/place_block.js

```diff
@@ -8,6 +8,14 @@ function inject (bot) {
     let oldBlock = bot.blockAt(dest)
     await bot._genericPlace(referenceBlock, faceVector, options)
 
+    // optimistic update
+    const heldItem = bot.heldItem
+    const mcData = require('minecraft-data')(bot.version)
+    const defaultState = mcData.blocksByName[heldItem?.name]?.defaultState
+    if (defaultState) {
+      bot.world.setBlockStateId(dest, defaultState)
+    }
```

Comment:
1. This should be using the registry
2. This is not a correct way to deduce correct block state from an item
3. If world state is mutated then you must ensure that the data is fully in sync with the server (how does this handle the case where the server rejects?)

Reply:
I believe I already answered 3rd question in the description. If the server rejects a block update would be emitted (even if it's air) as this is what the native client also does.

Can you elaborate on the 2nd question?